src/main/java/core/basesyntax/db/FruitDaoImpl.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax.db;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class FruitDaoImpl implements FruitDao {
+    private final Map<String, Integer> fruitDB;
+
+    public FruitDaoImpl() {
+        fruitDB = new HashMap<>();
+    }
+
+    @Override
+    public Integer putFruits(String key, Integer value) {
+        Integer numberOfFruits = fruitDB.get(key);
+
+        validateValue(numberOfFruits, value);
+        if (numberOfFruits == null) {
+            return fruitDB.put(key, value);
+        }
+        return fruitDB.put(key, numberOfFruits + value);
+    }
+
+    public Map<String, Integer> getFruitDB() {
```

Comment:
Exposing the internal data structure is against the encapsulation principle. Instead, provide methods that perform operations directly on the internal structure without giving access to it.